cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1311,12 +1312,18 @@ func gatherUnderlayNetwork(ds *datastore.RethinkStore, allocationSpec *machineAl
 
 func makeMachineNetwork(ds *datastore.RethinkStore, ipamer ipam.IPAMer, allocationSpec *machineAllocationSpec, n *allocationNetwork) (*metal.MachineNetwork, error) {
 	if n.auto {
+		allocationUUID, err := uuid.NewRandom()
+		if err != nil {
+			return nil, fmt.Errorf("unable to create uuid for IP allocation: %v", err)
+		}
+
 		ipAddress, ipParentCidr, err := allocateIP(n.network, "", ipamer)
 		if err != nil {
 			return nil, fmt.Errorf("unable to allocate an ip in network: %s %#v", n.network.ID, err)
 		}
 		ip := &metal.IP{
 			IPAddress:        ipAddress,
+			AllocationUUID:   allocationUUID.String(),
```

Comment:
This is not the only place where a IP is allocated, adding the allocationUUID must be done in a central place

Reply:
Mh, yes, that would be better. But it's also a bigger refactoring to make this look adequate... I am not sure, if this refactoring should be contained in this PR.

The IP create function of the IP service and the machine network auto ip allocation should be able to use a common `ipCreate` function that hands over an IPSpec as a parameter, I guess.

Reply:
TBH, this PR looks crazy as hell

Reply:
I can't deny that. But it looked simpler to me than other approaches in regards of time and suitability. I am open for ideas and comments. :)